### ISSUE TITLE
Add support for base/modifier cost and points distinctions

### ISSUE THREAD

**Issue by KingSupernova31** [KaiGeffen/Celestial#136]
Currently, card costs and points are handled inconsistently in many cases. Death says that its cost is 0, yet it's not removed by Secretary Bird. But Sunflower does take into account nourish when calculating its point value. Death displays as 0 in the hand, but as 7 in other zones. Pet says that it retains changes to its points as it resolves, but in reality it retains changes to its points no matter how it's changing zones. But other cards don't do that, and Pet doesn't display its points any differently than other cards. The proposed new Phoenix would have unclear interactions with everything else; Phoenix is worded the same way as nourish, but it supposed to persist across zones, so if you Phoenix a Pet and play it multiple times, does it increase in point cost every time you go through this loop? Etc.

This is a mess, and hard for me to keep track of even after hours of going through all the different cases and trying to figure out how they work. I expect it will rapidly become completely intractable if/when the card pool grows to a few hundred cards.

Proposed solution: A card's total cost is the sum of its base cost and any cost modifiers. Its total cost is what's displayed on the card while it's in your hand, and its base cost is what's displayed everywhere else. Its base cost is retained when it changes zones, while its cost modifiers reset.

The same is true for its points, except that base points are always displayed everywhere, total points are never displayed. (Personally I think total points should be displayed in the story, but that's a separate change that doesn't need to be discussed here. Most consistent with current functionality is that they're never displayed.)

I believe this would provide a consistent system under which all current cards can function almost the same way they do presently, while making it easy to add future cards like the proposed Phoenix and other stuff that wants to change/reference costs/points either permanently or temporarily.

This proposed system to my knowledge matches current functionality of all cards, so it doesn't necessitate any code changes right now. You'd just need to standardize card text so it's clearer what cards are doing. I think the most elegant way would be to let *changes* to a value be implied changes to a modifier, so a card can just say "this costs 0" or "worth +1 point" without needing to scare the player with the word "modifier". Then anything that specifically wants to change the base value, like the new Phoenix, uses the word "base" to make this clear. Anything that references a value by default uses the total, so again use the word "base" to clarify that it ignores modifications.

If this all sounds good, I've compiled a list of proposed new wordings to make things easy for you:

Night Vision: "Put the top card with base cost 2 from your deck into your hand."
Rebirth: "Transform each of your cards later in the story into a 0:X Fleeting Child, where X is its base cost."
Starling: "Worth +1 point if the next card in the story has base cost 1."
Fishing Boat: "Put the top 3 cards with base cost 1 from your deck into your hand."
Secretary Bird: "Worth +1 point for each card in your hand with base cost 0 or 1. Remove those cards from the game."
Cling: "Put the card with the highest base cost from your discard pile on top of your deck. Gain points equal to that card's cost."
New Phoenix: "Cards with fleeting in your discard pile and deck get +1 base points."
New Wing Clipping: "Your opponent discards 2 cards with the highest base cost from their hand."

**Comment by KaiGeffen** [comment]
With the merge of your card text rewrite, this has been implemented. A couple notes:
* Currently nothing modifies cost (Besides a card itself)
* Yes having a visual indicator of a card's points changing because of its effect / nourish would be great as a future feature
* Cards that have points different from the "normal" base-points for that card type have it displayed differently onscreen, but otherwise no text references this change
* New Phoenix is ambiguous because there isn't a way for Fleeting cards to resolve and stay in the game. So I think the card says in technical terms "Permanently change the base-points for each Fleeting card in your deck/discard pile", but right now that is the same as "Change the ..." for only a single resolution

**Comment by KingSupernova31** [comment]
I'm confused by your last bullet point, what's the ambiguity? The intention is that changes to the base cost always persist through zone changes, which is why Pet works at all. So "change the base points" and "permanently change the base points" are synonymous.

**Comment by KaiGeffen** [comment]
Yes! I meant to say that Phoenix could have either of the following 2 effects, which are effectively synonymous right now because cards with Fleeting will never resolve more than once:
1) Change the base-points of those Fleeting card permanently
2) Change the base-points of those Fleeting cards the next time they resolve
I agree with your suggestion that changes to base cost should always persist between zone-changes

**Comment by KingSupernova31** [comment]
Oh, I see. You're saying that a future effect might want to modify a card's points the next time it resolves, even if that's after a zone change, but not retain the modification after that resolution, and my proposed system doesn't handle that elegantly. This is an interesting point, I didn't consider that type of effect. If that effect is something you want to be common in the game, I can probably come up with a better system for it.

One simple option I suppose would be to have cost and point modifiers only reset when a card moves out of the story, and not reset on any other zone change. So an effect that changes a card in some other zone naturally persists until that card makes it into the story.


BTW I like the new Pet wording, I think that's much more elegant than the previous one. Under the hood we can think of it as saying "anything that would affect this card's points modifier affects its base points instead".

Also I think Rebirth and Secretary Bird missed the text change, they still just say "cost", not "base cost". The new Phoenix should also say "+1 base point", not "+1 point".
